.github/workflows/linuxtests.yml

```diff
@@ -0,0 +1,50 @@
+name: Linux Tests
+
+on:
+  push:
+    branches:
+      - master
+      - maintenance/**
+  pull_request:
+    branches:
+      - master
+      - maintenance/**
+
+jobs:
+  Python-37-dbg:
+    name: Python 3.7-dbg
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Configure env vars and python packages
+        run: |
+          sudo apt install python3.7-dbg python3.7-dev libatlas-base-dev liblapack-dev gfortran libgmp-dev libmpfr-dev libsuitesparse-dev ccache swig libmpc-dev
+          free -m
+          export PATH=/usr/lib/ccache:$PATH
+          ccache --zero-stats
+          export CCACHE_COMPRESS=1
+          python3.7-dbg --version # just to check
+          export NPY_NUM_BUILD_JOBS=2
+          uname -a
+          df -h
+          ulimit -a
+          set -e -o pipefail
+          mkdir builds
+          cd builds
+          python3.7-dbg -c 'import sys; print("Python debug build:", hasattr(sys, "gettotalrefcount"))'
+          python3.7-dbg -m pip install --upgrade pip setuptools wheel cython pytest pytest-xdist argparse pybind11
+          python3.7-dbg -m pip install --upgrade mpmath gmpy2
```

Comment:
is it necessary to have 3 `pip install --upgrade` lines, or can they be joined into 1?

Reply:
Apparently these ones require setuptools to be already installed. When put on the same line they start complaining that it is not there.  Same with the next numpy line